### PR TITLE
Clean up `save_checkpoint`

### PR DESCRIPTION
**Pull request by hanlint**
This PR cleans up the code in `save_checkpoint` for readability, and simplifies the logic.

### CODE REVIEW

File: composer/trainer/trainer.py [hunk 1]
```diff
@@ -2396,7 +2402,12 @@ def save_checkpoint(self, name: str = 'ep{epoch}-ba{batch}-rank{rank}', *, weigh
         Returns:
             str or None: See :func:`.save_checkpoint`.
         """
-        return save_checkpoint(state=self.state, filename=name, weights_only=weights_only)
+        return save_checkpoint(
```
Comment (eracah) [inline composer/trainer/trainer.py:2405]: any way we can fix both functions have the same name?

Comment (hanlint) [reply composer/trainer/trainer.py:2405]: I changed it to be:
```
def save_checkpoint(..):
    return checkpoint.save_checkpoint(...)
```

to clarify the save_checkpoint namespace a bit, does that work?

**Review by eracah — APPROVED** [review]
LGTM

**Review by mvpatel2000 — APPROVED** [review]
LGTM. I like the checkpoint.fn_name -- good idea

**Pull request merged**